### DISCUSSION TITLE
[Feature] dotnet caching in pipelines / nx cloud

### DISCUSSION THREAD

**Discussion by EelcoLos** [nx-dotnet/nx-dotnet#593] [Ideas]
**Is your feature request related to a problem? Please describe.**
When running `npm i` either locally or with pipelines the generator of nx-dotnet restore pops up
![image](https://user-images.githubusercontent.com/5102501/213918109-5c73648d-77b0-47ca-a660-bb2f0813bd0d.png)
On pipelines, this part of installing is the one not cached. This makes caching on my pipeline obsolete, as it will take as long to install via cache or normally (2m21 without cache (`npm i` took 1m16 of that) or 2m 29 (`npm i` took 1m19)).

**Describe the solution you'd like**
To have the dotnet packages being installed as normal dotnet projects described at https://learn.microsoft.com/en-us/azure/devops/pipelines/release/caching?view=azure-devops.

**Describe alternatives you've considered**
I'm considering removing the dotnet restore part of the postinstall/prepare option.
another thing i'm looking at is
```xml
<PropertyGroup>
    <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
  </PropertyGroup>
```
which might trigger the lockfile which the pipeline is referring to.

Otherwise, it would be nice that that `PropertyGroup` is created on creation

**Comment by AgentEnder** [comment]
Hey @EelcoLos! The step can definitely be removed from postinstall, you'll just want to run install prior to other tasks in the pipeline s.t. you can still have it available and use noRestore on builds and such (its a bit faster).

The main advantage to it being in postinstall is that the DTE setup with cloud is simpler, your not adding a second install step into the mix. But it can be moved.
